core/shared/src/main/scala/io/chrisdavenport/log4cats/StructuredLogger.scala

```diff
@@ -21,49 +21,54 @@ object StructuredLogger {
   def apply[F[_]](implicit ev: StructuredLogger[F]): StructuredLogger[F] = ev
 
   def withContext[F[_]](sl: StructuredLogger[F])(ctx: Map[String, String]): StructuredLogger[F] =
-    new ExtraContextStructuredLogger[F](sl)(ctx)
+    new ModifiedContextStructuredLogger[F](sl)(_ ++ ctx)
 
-  private class ExtraContextStructuredLogger[F[_]](sl: StructuredLogger[F])(
-      ctx: Map[String, String]
+  def withModifiedContext[F[_]](
+      sl: StructuredLogger[F]
+  )(modifyCtx: Endo[Map[String, String]]): StructuredLogger[F] =
+    new ModifiedContextStructuredLogger[F](sl)(modifyCtx)
+
+  private class ModifiedContextStructuredLogger[F[_]](sl: StructuredLogger[F])(
+      modify: Endo[Map[String, String]]
   ) extends StructuredLogger[F] {
-    private val outer = ctx
-    def error(message: => String): F[Unit] = sl.error(outer)(message)
-    def warn(message: => String): F[Unit] = sl.warn(outer)(message)
-    def info(message: => String): F[Unit] = sl.info(outer)(message)
-    def debug(message: => String): F[Unit] = sl.debug(outer)(message)
-    def trace(message: => String): F[Unit] = sl.trace(outer)(message)
+    private def whenEmpty: Map[String, String] = modify(Map.empty)
```

Comment:
This can be easily turned back into a `val`, right? Since applying a pure function to the same value, yields the same result.

Reply:
@lorandszakacs thanks! I've replaced with `lazy val`